drivers/video/hm01b0.c

```diff
@@ -284,6 +285,73 @@ static int hm01b0_soft_reset(const struct device *dev)
 	return ret;
 }
 
+static int hm01b0_set_frmival(const struct device *dev, struct video_frmival *frmival)
+{
+	const struct hm01b0_config *config = dev->config;
+	struct video_format fmt;
+	struct hm01b0_data *drv_data = dev->data;
+	int ret;
+
+	uint32_t osc_div = 0;
+	bool highres = false;
+
+	ret = hm01b0_get_fmt(dev, &fmt);
+	if (ret < 0) {
+		LOG_ERR("Can not get format!");
+		return ret;
+	}
+
+	if (fmt.width == 320) {
+		highres = true;
+	}
+
+
+	if (frmival->numerator <= 15) {
+		osc_div = highres ? 0x01 : 0x00;
+	} else if (frmival->numerator <= 30) {
+		osc_div = highres ? 0x02 : 0x01;
+	} else if (frmival->numerator <= 60) {
+		osc_div = highres ? 0x03 : 0x02;
+	} else {
+		/* Set to the max possible FPS at this resolution. */
+		osc_div = 3;
+	}
```

Comment:
This is applying a `ceil()` to the frame interval from the user, it's also possible to use `void video_closest_frmival()`
for getting the closest match, and then you can check with `==`.

https://github.com/nxp-upstream/zephyr/blob/40f07a0532ed1ddabf33180eb2be0618bc19e832/include/zephyr/drivers/video.h#L933-L950

However, this requires also supporting `video_enum_frmival()`, which is mandatory for supporting frame intervals.

Given you are one of the first to implement this API on an image sensor there are not a lot of examples for how to do it. You could try, like suggested in another PR by Phi Bang, without ` = 60/30/15`:

```c
enum {
		HM01B0_60_FPS,
		HM01B0_30_FPS,
		HM01B0_15_FPS,
};
```

Then here:

```suggestion
	struct video_frmival_enum fie = {.format = &data->fmt, .discrete = *frmival};

	video_closest_frmival(dev, &fie);
	
	switch (fie.index) {
	case HM01B0_60_FPS:
		osc_div = highres ? 0x01 : 0x00;
		break;
	case HM01B0_30_FPS:
		osc_div = highres ? 0x02 : 0x01;
		break;
	case HM01B0_15_FPS:
		osc_div = highres ? 0x03 : 0x02;
		break;
	default:
		LOG_ERR("Unsupported frame interval value");
		return -ENOTSUP;
	}
```

Reply:
And to implement `hm01b0_enum_frmival()` I think this might work:

```c
int hm01b0_enum_frmival(const struct device *dev, struct video_frmival_enum *fie)
{
	switch (fie->index) {
	case HM01B0_60_FPS:
		fie->discrete.denominator = 60;
		break;
	case HM01B0_30_FPS:
		fie->discrete.denominator = 30;
		break;
	case HM01B0_15_FPS:
		fie->discrete.denominator = 15;
		break;
	default:
		return -EINVAL;
	}

	fie->type = VIDEO_FRMIVAL_TYPE_DISCRETE;
	fie->discrete.numerator = 1;

	return 0;
}
```

Reply:
@josuah 
@KurtE  - pointed out that you suggested this change, since I have the same in the HM0360 will update that after I fix up my pr issues.

Thanks guys

Reply:
done, I also added 4th for lower speed.